registry/client_test.go

```diff
@@ -14,6 +15,43 @@ import (
 	tfversion "github.com/hashicorp/terraform/version"
 )
 
+func TestConfigureRegistryClientTimeout(t *testing.T) {
+	t.Run("default timeout", func(t *testing.T) {
+		if requestTimeout != defaultRequestTimeout {
+			t.Fatalf("expected timeout %q, got %q",
+				defaultRequestTimeout.String(), requestTimeout.String())
+		}
+
+		rc := NewClient(nil, nil)
+		if rc.client.Timeout != defaultRequestTimeout {
+			t.Fatalf("expected client timeout %q, got %q",
+				defaultRequestTimeout.String(), rc.client.Timeout.String())
+		}
+	})
+
+	t.Run("configured timeout", func(t *testing.T) {
+		defer func() {
+			os.Setenv(registryClientTimeoutEnvName,
+				os.Getenv(registryClientTimeoutEnvName))
+			requestTimeout = defaultRequestTimeout
+		}()
```

Comment:
I may be misunderstanding what's happening here, but I think this `Setenv(Getenv)` call won't do anything. A common pattern I've seen is `defer os.Setenv(os.Getenv(something))`, but wrapping it in a closure will prevent evaluation of the `Getenv` until it's too late. Untested alternative:

```suggestion
		defer func(timeoutEnv) {
			os.Setenv(registryClientTimeoutEnvName, timeoutEnv)
			requestTimeout = defaultRequestTimeout
		}(os.Getenv(registryClientTimeoutEnvName))
```

Reply:
The arguments to a deferred function are evaluated immediately (it's "outside" the closure), so `Getenv` here should be ok.

Reply:
good catch! overlooked this cleanup. going to commit your change here and update the other PR which has the same issue too.

Reply:
Sorry about the confusing comment here, I missed "suggested change" and though this was the highlighted snippet. This is correct 👍